sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -11,6 +12,27 @@ namespace Azure.Core
     /// </summary>
     public abstract class TokenCredential
     {
+        /// <summary>
+        /// Create the instance of a <see cref="TokenCredential"/>.
+        /// </summary>
+        protected TokenCredential() : this (TimeSpan.FromMinutes(2)) { }
+
+        /// <summary>
+        /// Create the instance of a <see cref="TokenCredential"/>.
+        /// </summary>
+        /// <param name="tokenRefreshOffset">A <see cref="TimeSpan"/> value representing the amount of time to subtract from the token expiry time, whereupon attempts will be made to refresh the token.</param>
+        protected TokenCredential(TimeSpan tokenRefreshOffset)
+        {
+            TokenRefreshOffset = tokenRefreshOffset;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> value representing the amount of time to subtract from the token expiry time, whereupon
+        /// attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the token.
+        /// </summary>
+        /// <returns>The duration value representing the amount of time to subtract from the token expiry time.</returns>
+        protected internal TimeSpan TokenRefreshOffset { get; }
```

Comment:
I was originally thinking this property could be virtual and the default implementation would return the default value. This would avoid the need for another protected constructor, and possibly provide a more consistent mocking story. Thoughts?

Reply:
The problem with `default` value is that default for `TimeSpan` is 2 minutes. We either have to make default value `null` and make the property nullable, or override default value internally, which isn't obvious for user.

Reply:
I didn't mean default as in the C# keyword. I meant the default value of two minutes. Sorry for the ambiguity.

Reply:
Property is now `public virtual` and can be overridden in derived types